src/hooks/useFormFields.ts

```diff
@@ -1,35 +1,45 @@
 import { useMemo } from 'react';
-import { type FormSchema, type FormField } from '../types';
+import { type FormField, type FormSchema } from '../types';
 
 export function useFormFields(form: FormSchema): { formFields: FormField[]; conceptReferences: Set<string> } {
   const [flattenedFields, conceptReferences] = useMemo(() => {
-    const flattenedFieldsTemp = [];
+    const flattenedFieldsTemp: FormField[] = [];
     const conceptReferencesTemp = new Set<string>();
+
+    const flattenFields = (fields: FormField[]) => {
```

Comment:
This function is recursive and causes side effects to external states, making it less pure. Below is an example of how we can simplify this logic (AI generated)

```TS

const processFlattenedFields = (fields: FormField[]): {
      flattenedFields: FormField[];
      conceptReferences: Set<string>;
    } => {
      const flattenedFields: FormField[] = [];
      const conceptReferences = new Set<string>();

      const processField = (field: FormField, parentGroupId?: string) => {
        // Add group ID to nested fields if applicable
        const processedField = parentGroupId 
          ? { ...field, meta: { ...field.meta, groupId: parentGroupId } }
          : field;

        // Add field to flattened list
        flattenedFields.push(processedField);

        // Collect concept references
        if (processedField.questionOptions?.concept) {
          conceptReferences.add(processedField.questionOptions.concept);
        }

        // Collect concept references from answers
        processedField.questionOptions?.answers?.forEach((answer) => {
          if (answer.concept) {
            conceptReferences.add(answer.concept);
          }
        });

        // Recursively process nested questions for obsGroup
        if (processedField.type === 'obsGroup' && processedField.questions) {
          processedField.questions.forEach((nestedField) => {
            processField(nestedField, processedField.id);
          });
        }
      };

      // Process all input fields
      fields.forEach(processField);

      return { flattenedFields, conceptReferences };
    };
    
  ```